ssz-rs/src/merkleization/mod.rs

```diff
@@ -145,9 +155,17 @@ fn merkleize_chunks_with_virtual_padding(
                 Ordering::Equal => {
                     let focus =
                         &mut layer[parent_index * BYTES_PER_CHUNK..(i + 1) * BYTES_PER_CHUNK];
+                    // checked subtraction is unnecessary:
+                    // focus.len() = (i + 1 - parent_index) * BYTES_PER_CHUNK
+                    // and
+                    // i >= parent_index
+                    // so focus.len() >= BYTES_PER_CHUNK; qed
                     let children_index = focus.len() - BYTES_PER_CHUNK;
-                    let (parent, children) = focus.split_at_mut(children_index);
-                    let (left, _) = children.split_at_mut(BYTES_PER_CHUNK);
+                    let (parent, left) = focus.split_at_mut(children_index);
```

Comment:
this will fail if `left` happens to be more than _just_ 32 bytes... let's verify before making this simplification